File: Sliding Window/Minimum Window SubString/Brute/code.cpp
#include <bits/stdc++.h>
using namespace std;

bool containsAll(string sub, string t) {
    vector<int> count(128, 0);
    for (char c : sub) count[c]++;
    for (char c : t) if (--count[c] < 0) return false;
    return true;
}

string minWindowBrute(string s, string t) {
    int n = s.size(), minLen = INT_MAX;
    string ans = "";
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            string sub = s.substr(i, j - i + 1);
            if (containsAll(sub, t) && sub.size() < minLen) {
                ans = sub;
                minLen = sub.size();
            }
        }
    }
    return ans;
}

int main() {
    string s = "ADOBECODEBANC", t = "ABC";
    cout << minWindowBrute(s, t) << endl; // Expected: BANC
}

File: Sliding Window/Minimum Window SubString/Brute/code.py
from collections import Counter

def containsAll(sub, t):
    count = Counter(sub)
    for c in t:
        count[c] -= 1
        if count[c] < 0:
            return False
    return True

def minWindowBrute(s, t):
    n = len(s)
    ans, minLen = "", float("inf")
    for i in range(n):
        for j in range(i, n):
            sub = s[i:j+1]
            if containsAll(sub, t) and len(sub) < minLen:
                ans, minLen = sub, len(sub)
    return ans

if __name__ == "__main__":
    s, t = "ADOBECODEBANC", "ABC"
    print(minWindowBrute(s, t))  # Expected: BANC
